include/swift/Driver/DependencyGraph.h

```diff
@@ -121,6 +121,13 @@ class DependencyGraphImpl {
   llvm::StringMap<std::pair<std::vector<DependencyEntryTy>, DependencyMaskTy>> Dependencies;
 
   /// The set of marked nodes.
+  /// "Marked" means that everything provided by this node (i.e. Job) is
+  /// dirty. Thus any file using any of these provides must be recompiled.
+  /// (Only non-private entities are output as provides.)
+  /// In other words, this Job "cascades"; the need to recompile it causes other
+  /// recompilations. It is possible that the current code marks things that do
+  /// not need to be marked. Nothing would break if that were the case.
```

Comment:
"Nothing would break" seems a bit too generic; the effect of over-marking is unnecessary recompilation.